src/native/managed/cdacreader/src/Contracts/Metadata_1.cs

```diff
@@ -0,0 +1,337 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection.Metadata.Ecma335;
+using Microsoft.Diagnostics.DataContractReader.Data;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts;
+
+// GC Heap corruption may create situations where a putative pointer to a MethodTable
+// may point to garbage. So this struct represents a MethodTable that we don't necessarily
+// trust to be valid.
```

Comment:
None of the data in the target process are trusted to be valid, even after they are validated.

I understand what you are trying to say, but "trust" is probably not the best word to describe it since it has security connotations.

Naming idea: `NonValidatedMethodTable`?